File: 940-fruit-into-baskets/fruit-into-baskets.java
class Solution {
    public int totalFruit(int[] fruits) {
        int l=0;
        int res=0;
        int n=fruits.length;
        Map<Integer,Integer>mp=new HashMap<>();
        for(int i=0;i<n;i++){
            mp.put(fruits[i],mp.getOrDefault(fruits[i],0)+1);
            while(mp.size()>2){
                mp.put(fruits[l],mp.get(fruits[l])-1);
                if(mp.get(fruits[l])==0)mp.remove(fruits[l]);
                l++;
            }
            res=Math.max(res,i-l+1);
        }
        return res;

    }
}

File: 940-fruit-into-baskets/fruit-into-baskets.cpp
class Solution {
public:
    int totalFruit(vector<int>& fruits) {
        int count=0;
        int n=fruits.size();
        map<int,int>mp;
        int l=0;
        for(int i=0;i<n;i++){
            mp[fruits[i]]++;
            while(mp.size()>2){
                mp[fruits[l]]--;
                if(mp[fruits[l]]==0)mp.erase(fruits[l]);
                l++;
            }
            count=max(count,i-l+1);
        }return count;
    }
};